dashboard/views.py

```diff
@@ -625,6 +641,57 @@ def are_weighted_assignments_hidden(course_id, df):
             return True
 
 
+def find_binning_grade_value(grades):
+    fifth_item = grades[4]
+    next_to_fifth_element = grades[5]
+    if next_to_fifth_element - fifth_item > 2:
+        binning_grade_value, index = fifth_item, 4
+    else:
+        binning_grade_value, index = binning_logic(grades, fifth_item)
+    return binning_grade_value, index
+
+
+def is_odd(num):
+    if num % 2 == 0:
+        return False
+    else:
+        return True
+
+
+def check_if_grade_qualify_for_binning(grade, fifthElement):
+    # case: 96.7, 94.76,
+    if int(grade) - int(fifthElement) > 1:
+        return False
+    # case: 94.86, 94.76
+    if int(grade) - int(fifthElement) == 0:
+        return True
+    # case 95.89, 94.76
+    if is_odd(int(grade)):
+        return True
+
+
+def binning_logic(grades, fifth_item_in_list):
+    """
+    Histogram binning is by 2 [ [0,2], [2,4], [4,6], …..] each item in the list starting number is inclusive and second
+    is exclusive.
+    Case 1: Just last 5 are binned
+    Actual distribution: [69.79, 80.0, 80.5, 88.21, 88.79, 92.71, 92.71, 92.71, 93.14, 94.43]
+    Binning Distribution: [88.79, 88.79, 88.79, 88.79, 88.79, 92.71, 92.71, 92.71, 93.14, 94.43]
+    Case 2: More than last 5 are binned based on histogram binning by count of 2
+    Actual Distribution: [90.77, 93.09, 93.42, 94.85, 94.87, 94.88, 94.9, 95.55, 95.89, 96.28, 96.4, 96.47, 96.49, 96.68]
+    Binning Distribution: [95.89, 95.89, 95.89, 95.89, 95.89, 95.89, 95.89, 95.89, 95.89,96.28, 96.4, 96.47, 96.49, 96.68]
+
+    :param grades: sorted in asc
+    :param fifth_item_in_list:
+    :return: max grade in the binned list, length of binned grades
+    """
+    binning_list=grades[:5]
+    for grade in grades[5:]:
+        if check_if_grade_qualify_for_binning(grade, fifth_item_in_list):
+            binning_list.append(grade)
+    return max(binning_list), len(binning_list)
```

Comment:
Seems like it's unnecessary to check all grades above the fifth; if the grades are sorted by value, couldn't you just use a `return` once `check_if_grade_qualify_for_binning` returns `False`?

Reply:
Yes, I agree. I will return on First False.

Reply:
Done